blog/2024-08-22-wasmcloud-on-the-factory-floor/index.mdx

```diff
@@ -0,0 +1,128 @@
+
+### Architectural Freedom = Resource Efficiency
+
+In traditional IoT models, processing is linear and unidirectional. Data is collected from separate sensors: captured by an edge, then ingested into a cloud—where the majority of compute takes place—before being handed off to the consumer. By pushing processing to the cloud, higher costs are incurred and latency increases. More importantly, valuable, more efficient processing capacity at the edge is underutilized.
+
+MachineMetrics does the majority of its data processing in Kubernetes in the cloud. By moving compute to devices, existing edge resources are used more effectively and cloud resources preserved. Porting logic between resources means the team can balance their resources against their needs. They use the cloud to scale the processing of larger data sets, whilst allowing edge devices to process and stream real-time data directly to operators.
+
+Jochen Rau: 
+
+> “The focus is to find ways to more efficiently use resources that we already have at the edge, and manage our cloud resources more effectively. wasmCloud helps us do that. It’s also a balance—edge devices are relatively small so we can scale much more easily in the cloud. Having the flexibility to move workloads around means greater resource efficiency.”*
+
+### Extending the Value of Existing Architecture
+
+MachineMetrics uses Kubernetes extensively, but the team can’t run it at the edge. This is partly due to the company’s OS configuration, but primarily due to the limitations of Kubernetes. Kubernetes is great at managing infrastructure, but not so good at running applications on resource-constrained devices.
+
+To solve this problem, [MachineMetrics](https://www.youtube.com/watch?v=fQdkNGZqYZA) uses wasmCloud alongside Kubernetes and ArgoCD, making use of the [wasmcloud-operator](https://github.com/wasmcloud/wasmcloud-operator) to deploy software to all of its edge locations. The wasmcloud-operator allows platform engineers to manage and run Wasm on Kubernetes using the standard, familiar controller pattern and custom resource definitions (CRDs), all while remaining decoupled from Kubernetes and free to leverage the unique benefits of components. Bringing the kind of extensibility to Kubernetes that was previously impossible.
+
+Rau says: 
+
+> “We shifted up a level: taking the wasmCloud host and deploying it on Kubernetes. The wasmcloud-operator makes this really simple to manage with our existing tools. We deploy our compute workloads on the lattice on wasmCloud. We’re crossing the boundaries between edge and cloud.”
+
+Using this “decoupled” model means it’s easy for MachineMetrics to tie wasmCloud directly into existing pipelines and tools like ArgoCD.
+
+![wasmCloud Kubernetes integration](./images/mm-4.png)
```

Comment:
I like this style. Would it be useful to have a `<Callout>...</Callout>` component that generates text like this without you having to build an image each time?

For now, alt text here should be the same as the text in the image.

```suggestion
!["WebAssembly, wasmCloud, and NATS will not only reshape the MachineMetrics business, but are already transforming industrial IoT."](./images/mm-4.png)
```

Reply:
Oop, missed it. No matter. I'll whip that component up anyway.